js/components/historical-navigation.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import IComponentProps from '../interfaces/IComponentProps';
+import registeredComponents from '../constants/registeredComponents';
+
+import '../../css/historical-navigation.less';
+
+declare const manywho: any;
+
+const navigateToPath = (flowKey, path) => {
+    manywho.engine.navigate(null, null, null, flowKey, path);
+};
+
+/**
+ * @description Renders a list of navigation links to previously visited map elements
+ */
+const HistoricalNavigation:React.FC<IComponentProps> = ({ flowKey }) => {
+
+    const [expanded, setExpanded] = React.useState(false);
+
+    const navigation = manywho.model.getHistoricalNavigation(flowKey);
+
+    const onEntryClick = (entry) => {
+        setExpanded(false); 
+        navigateToPath(flowKey, entry.path);
+    };
+
+    if (navigation && navigation.entries && navigation.entries.length > 0) {
+
+        return (
+            <nav className={`historical-navigation${expanded ? ' expanded' : ''}`}>
+                <ul>
+                    {
+                        navigation.entries.map((entry, index) => (
+                            <li key={index}>
+                                <button className="btn btn-link" onClick={() => onEntryClick(entry)}>
+                                {
+                                    entry.mapElementName
+                                }
+                                </button>
+                                {
+                                    index === 0 ? (
+                                        <button 
```

Comment:
What do you think about giving this the title "expand" so that people are told what it does?

Reply:
Good idea. Done.